packages/test/test-end-to-end-tests/src/test/gc/gcSweepAttachmentBlobs.spec.ts

```diff
@@ -0,0 +1,1056 @@
+				// that the blob is requested via its url since this container does not have access to the blob's handle.
+				const container2 = await loadContainer(summary2.summaryVersion);
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle1.absolutePath,
+					"Container2: Blob1",
+				);
+				await validateBlobRetrievalFails(
+					container2,
+					blobHandle2.absolutePath,
+					"Container2: Blob2",
+				);
+
+				// Retrieving the blobs in the summarizer container should fail as well.
+				await validateBlobRetrievalFails(
+					summarizerContainer,
+					blobHandle1.absolutePath,
+					"Summarizer: Blob1",
+				);
+				await validateBlobRetrievalFails(
+					summarizerContainer,
+					blobHandle2.absolutePath,
+					"Summarizer: Blob2",
+				);
+			},
+		);
+
+		/**
+		 * This test validates that when blobs are de-duped in different containers, these containers can use these
+		 * blobs irrespective of whether the original blob is deleted. Basically, after uploading a blob, a container
+		 * should be able to use it the same way whether it was de-duped or not.
+		 */
+		it("should allow access to blobs that are de-duped in different containers", async () => {
+			const { dataStore: mainDataStore, summarizer } = await createDataStoreAndSummarizer();
+
+			// Upload an attachment blob.
+			const blobContents = "Blob contents";
+			const blobHandle = await mainDataStore._runtime.uploadBlob(
+				stringToBuffer(blobContents, "utf-8"),
+			);
+
+			// Reference and then unreference the blob so that it's unreferenced in the next summary.
+			mainDataStore._root.set("blob1", blobHandle);
+			mainDataStore._root.delete("blob1");
+
+			// Summarize so that the above attachment blobs are marked unreferenced.
+			await provider.ensureSynchronized();
+			const summary1 = await summarizeNow(summarizer);
+
+			// Wait for half sweep timeout and load a container. This container will upload a blob with the same content
+			// as above so that it is de-duped. This container should be able to use this blob until its session
+			// expires.
+			await delay(sweepTimeoutMs / 2);
+			const container2 = await loadContainer(summary1.summaryVersion);
+			const container2MainDataStore = await requestFluidObject<ITestDataObject>(
+				container2,
+				"default",
+			);
+			// Upload the blob and keep the handle around until the blob uploaded by first container is deleted.
+			const container2BlobHandle = await container2MainDataStore._runtime.uploadBlob(
+				stringToBuffer(blobContents, "utf-8"),
+			);
+
+			// Wait for sweep timeout so that the blob uploaded by the first container is ready to be deleted.
+			await delay(sweepTimeoutMs / 2 + 10);
+
+			// Send an op to update the current reference timestamp that GC uses to make sweep ready objects.
+			mainDataStore._root.set("key", "value");
+			await provider.ensureSynchronized();
+
+			// Summarize so that the blob is now deleted.
+			const summary2 = await summarizeNow(summarizer);
```

Comment:
If I'm correct, the original blob should not be deleted right? Because we uploaded the blob again as container2BlobHandle which caused a dedupe. I wonder if we should do a check here with container2BlobHandle.get();

Reply:
container3BlobHandle.get() does the same thing. But adding this check does make sense. I will add it.

Reply:
done